rep-2014.rst

```diff
@@ -0,0 +1,242 @@
+REP: 2014
+Title: Benchmarking performance in ROS 2
+Author: Víctor Mayoral-Vilches <victor@accelerationrobotics.com>, Ingo Lütkebohle <Ingo.Luetkebohle@de.bosch.com>, Christophe Bédard <bedard.christophe@gmail.com>
+Status: Draft
+Type: Informational
+Content-Type: text/x-rst
+Created: 29-Sept-2022
+Post-History: 05-Nov-2022
+
+
+Abstract
+========
+
+This REP describes some principles and guidelines for benchmarking performance in ROS 2.
+
+
+Motivation
+==========
+
+Benchmarking is the act of running a computer program with a known workload to assess the program's relative performance. In the context of ROS 2, performance information can help roboticists design more efficient robotic systems and select the right hardware for their robotic application. It can also help understand the trade-offs between different algorithms that implement the same capability, and help them choose the best approach for their use case. Performance data can also be used to compare different versions of ROS 2 and to identify regressions. Finally, performance information can be used to help prioritize future development efforts.
+
+
+The myriad combinations of robot hardware and robotics software make assessing robotic-system performance in an architecture-neutral, representative, and reproducible manner challenging. This REP attempts to provide some guidelines to help roboticists benchmark their systems in a consistent and reproducible manner by following a quantitative approach. This REP also provides a set of tools and examples to help guide roboticists while collecting and reporting performance data.
+
+Value for stakeholders:
+
+- Package maintainers can use these guidelines to integrate performance benchmarking tools (e.g. instrumentation) and data (e.g. results, plots and datasets) in their packages.
+
+- Consumers can use the guidelines in the REP to benchmark ROS Nodes and Graphs in an architecture-neutral, representative, and reproducible manner, as well as the corresponding performance data offered in ROS packages to set expectations on the capabilities of each.
+
+- Hardware vendors and robot manufacturers can use these guidelines to show evidence of the performance of their systems solutions with ROS in an architecture-neutral, representative, and reproducible manner.
+
+The guidelines in here are intended to be a living document that can be updated as new information becomes available.
+
+Tracing and benchmarking
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+Tracing and benchmarking can be defined as follows:
+
+- `tracing`: logging (partial) execution information while the system is running. `tracing` is used to understand what goes on in a running software system. 
```

Comment:
Do we have scope for distributed system with this REP? It looks this PR is only for localhost system?

```suggestion
- `tracing`: logging (partial) execution information while the system is running. `tracing` is used to understand what goes on in running software systems. 
```